db/sql/sql.go

```diff
@@ -686,10 +734,19 @@ func (tx *Transaction) UpdateContext(ctx context.Context, resource *schema.Resou
 	return tx.exec(ctx, sql, args...)
 }
 
+func (tx *CachedTransaction) StateUpdate(resource *schema.Resource, state *transaction.ResourceState) error {
+	return tx.StateUpdateContext(context.Background(), resource, state)
+}
+
 func (tx *Transaction) StateUpdate(resource *schema.Resource, state *transaction.ResourceState) error {
 	return tx.StateUpdateContext(context.Background(), resource, state)
 }
 
+func (tx *CachedTransaction) StateUpdateContext(ctx context.Context, resource *schema.Resource, state *transaction.ResourceState) error {
+	tx.ClearCache()
+	return tx.StateUpdateContext(context.Background(), resource, state)
```

Comment:
infinite recursion: should be `tx.TxInterface.StateUpdateContext`

Reply:
right, will fix.